model/src/main/proto/exploration.proto

```diff
@@ -319,22 +336,40 @@ message HelpIndex {
   // This type is uninitialized in cases when no index is currently available (either because a hint
   // or solution hasn't yet been triggered, or because there are none to trigger).
   oneof index_type {
-    // Indicates this help index corresponds to the index of a hint within the hint list of a state.
-    int32 hint_index = 1;
+    // Indicates this help index corresponds to the index of the next available hint within the hint
+    // list of a state.
+    int32 available_next_hint_index = 1;
```

Comment:
In general we shouldn't do modifications to protos like this (see more here https://developers.google.com/protocol-buffers/docs/proto3#updating) . Can we instead deprecate hint_index and add the new fields in the end? i.e.
```
reserved 1;

// ...
bool show_solution = 2;

// ...
bool everything_revealed = 3;

// ...
int32 available_next_hint_index = 4;

// ...
int32 latest_revealed_hint_index = 5;
```

Reply:
done

Reply:
This appears to be resolved (mainly since the old index has now been reserved). Closing the thread with permission from @vinitamurthi to verify & close out old conversation threads.